### PR TITLE
[WIP] Update/refined 0.9.8

### PR DESCRIPTION
**Pull request by avdv**

### CODE REVIEW

**Comment by kwark** [comment]
There is no version of play yet that supports 2.13.0, so I can't merge this for now.
https://github.com/playframework/playframework/issues/9260 tracks the 2.13 progress for play.
It might that play will only be available for scala 2.13.0 from version 2.8.0 onwards.

**Comment by avdv** [comment]
> There is no version of play yet that supports 2.13.0, so I can't merge this for now.

Yeah, I just meant this PR to be work-in-progress. Sorry, should have mentioned that in the description.

**Comment by avdv** [comment]
The ball is rolling again. There's not an official announcement yet, but get ready...

**Comment by kwark** [comment]
https://github.com/playframework/scalatestplus-play/pull/167 tracks the scala 2.13.0 release

File: project/Dependencies.scala [hunk 1]
```diff
@@ -26,14 +26,16 @@ object Dependencies {
...
+  val play27Json          = "com.typesafe.play"          %%    "play-json"                % "2.7.4"
   val play27              = "com.typesafe.play"          %%    "play"                     % play27Version
   val play27NettyServer   = "com.typesafe.play"          %%    "play-netty-server"        % play27Version  % "test"
   val scalaTestPlusPlay27 = "org.scalatestplus.play"     %%    "scalatestplus-play"       % "4.0.1"  % "test"
 
+  val play27NettyUnix     = "io.netty" % "netty-transport-native-unix-common" % "4.1.34.Final" % "test"
+
   val testDependencies = Seq(scalaCheck)
-  val play27Dependencies = Seq(refined, play27, play27Json, play27NettyServer, scalaTestPlusPlay27, play27WsJson)
+  val play27Dependencies = Seq(refined, play27, play27Json, play27NettyServer, scalaTestPlusPlay27, play27WsJson, play27NettyUnix)
```
Comment (avdv) [inline project/Dependencies.scala]: I have no idea why this is now necessary, the dependency should have been added transitively by `play-netty-server`.

**Comment by avdv** [comment]
:champagne: Tests were green.

But, I also noticed some errors during the different test runs:

1) for Scala 2.12

```
sbt:root> test
[info] Compiling 1 Scala source to /home/claudio/github/play-refined/play26-refined/target/scala-2.12/test-classes ...
[info] Compiling 1 Scala source to /home/claudio/github/play-refined/play27-refined/target/scala-2.12/test-classes ...
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[info] SIRDRoutingSuite:
[info] - refined string routing
[info] - refined string query
[info] - refined int routing
[info] - refined int query
[info] - refined long routing
[info] - refined long query
[info] - refined double routing
[info] - refined double query
[error] /play-refined/src/test/scala/be/venneborg/refined/play/SIRDServerSuite.scala:24:42: exception during macro expansion: 
[error] java.lang.AssertionError: assertion failed: import failure: cannot determine unique overloaded method alternative from
[error]  final package shapeless
[error] final package shapeless
[error]  that matches package shapeless:shapeless.type
[error] 	at scala.reflect.internal.SymbolTable.throwAssertionError(SymbolTable.scala:183)
[error] 	at scala.reflect.internal.Importers$StandardImporter.disambiguate$1(Importers.scala:195)
[error] 	at scala.reflect.internal.Importers$StandardImporter.recreateOrRelink$1(Importers.scala:200)
[error] 	at scala.reflect.internal.Importers$StandardImporter.importSymbol(Importers.scala:222)
[error] 	at scala.reflect.internal.Importers$StandardImporter.recreateType(Importers.scala:240)
[error] 	at scala.reflect.internal.Importers$StandardImporter.importType(Importers.scala:296)
[error] 	at scala.reflect.internal.Importers$StandardImporter.recreateType(Importers.scala:240)
[error] 	at scala.reflect.internal.Importers$StandardImporter.importType(Importers.scala:296)
[error] 	at scala.reflect.internal.Importers$StandardImporter.recreateType(Importers.scala:240)
[error] 	at scala.reflect.internal.Importers$StandardImporter.importType(Importers.scala:296)
[error] 	at scala.reflect.internal.Importers$StandardImporter.recreateType(Importers.scala:236)
[error] 	at scala.reflect.internal.Importers$StandardImporter.importType(Importers.scala:296)
[error] 	at scala.reflect.internal.Importers$StandardImporter.$anonfun$recreateType$1(Importers.scala:236)
[error] 	at scala.reflect.internal.Importers$StandardImporter.recreateType(Importers.scala:236)
[error] 	at scala.reflect.internal.Importers$StandardImporter.importType(Importers.scala:296)
[error] 	at scala.reflect.internal.Importers$StandardImporter.recreatedTreeCompleter(Importers.scala:314)
```

2) for all Scala versions
```
[info] SIRDServerSuite:
[info] be.venneborg.refined.play.SIRDServerSuite *** ABORTED ***
[info]   play.core.server.ServerListenException: Failed to listen for HTTP on /0.0.0.0:9000!
[info]   at play.core.server.NettyServer.bindChannel(NettyServer.scala:231)
[info]   at play.core.server.NettyServer.$anonfun$httpChannel$1(NettyServer.scala:219)
[info]   at play.core.server.NettyServer.$anonfun$httpChannel$1$adapted(NettyServer.scala:219)
[info]   at scala.Option.map(Option.scala:163)
[info]   at play.core.server.NettyServer.<init>(NettyServer.scala:219)
[info]   at play.core.server.NettyServerComponents.server(NettyServer.scala:371)
[info]   at play.core.server.NettyServerComponents.server$(NettyServer.scala:367)
[info]   at play.core.server.NettyServer$$anon$1.server$lzycompute(NettyServer.scala:356)
[info]   at play.core.server.NettyServer$$anon$1.server(NettyServer.scala:356)
[info]   at play.core.server.NettyServer$.createServerFromRouter(NettyServer.scala:356)
[info]   ...
```
This happens because starting the single `test` task, kicks of its child tasks in parallel, so `play-refined25`, `play-refined26` and `play-refined27` are running concurrently. This clashes because every one of them wants to start the SIRD server and bind it to port 9000.

I tried to set `Global / concurrentRestrictions += Tags.limit(Tags.Test, 1)` as documented [here](https://www.scala-sbt.org/1.x/docs/Parallel-Execution.html) but to no avail. A simple solution would be to run those tasks sequentially by explicitly listing them to SBT. In general, it would be much better to use an ephemeral port - don't know whether that is possible.

**Comment by kwark** [comment]
@avdv Thanks Claudio for this PR. 🎉 
https://github.com/kwark/play-refined/issues/51 tracks the issue with SIRDServerSuite sometimes failing on Travis, but I was under the impression that on Travis the builds and tests for different scala versions run on different VMs.

**Comment by avdv** [comment]
Hi Peter,

 > #51 tracks the issue with SIRDServerSuite sometimes failing on Travis, but I was under the impression that on Travis the builds and tests for different scala versions run on different VMs.

Yes, the jobs on Travis run per Scala version according to the build matrix. What I was referring to was that I had seen the first error only in Scala 2.12 jobs, and the second problem turned up randomly in any job, regardless of the Scala version.

**Pull request merged**